### PR TITLE
Add Strategy Design Pattern

### PR DESCRIPTION
**Pull request by aquaruiz**

### CODE REVIEW

**Comment by aquaruiz** [comment]
@qnislavnachev

File: design-patterns/behavioral-patterns/src/strategy/Strategy.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package strategy;
+
+public interface Strategy {
+    int executeStrategy(int a, int b);
```
Comment (qnislavnachev) [inline design-patterns/behavioral-patterns/src/strategy/Strategy.java]: execute() is enough I believe.

File: design-patterns/behavioral-patterns/src/strategy/Strategy.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package strategy;
+
+public interface Strategy {
```
Comment (qnislavnachev) [inline design-patterns/behavioral-patterns/src/strategy/Strategy.java]: MathStrategy is better name for this interface I think

File: design-patterns/behavioral-patterns/src/strategy/ConcreteStrategyMultiply.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package strategy;
+
+public class ConcreteStrategyMultiply implements Strategy {
```
Comment (qnislavnachev) [inline design-patterns/behavioral-patterns/src/strategy/ConcreteStrategyMultiply.java]: MultiplicationStrategy will fit better as a name. Apply same naming convention for the rest strategies.

**Review by qnislavnachev — CHANGES_REQUESTED** [review]

**Pull request merged**